src/v/storage/segment.cc

```diff
@@ -342,6 +343,9 @@ ss::future<> segment::do_flush() {
         _tracker.committed_offset = std::max(o, _tracker.committed_offset);
         _tracker.stable_offset = _tracker.committed_offset;
         _reader->set_file_size(std::max(fsize, _reader->file_size()));
+        if (_cache) {
+            _cache->mark_clean(_tracker.stable_offset);
+        }
```

Comment:
is this not already handled by mark_clean handling in advance_stable_offset?

Reply:
It is. Removed it.